lrauv_ignition_plugins/proto/dvl_velocity_tracking.proto

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package lrauv_ignition_plugins.msgs;
+option java_package = "lrauv_ignition_plugins.msgs";
+option java_outer_classname = "DVLProtos";
+
+/// \ingroup lrauv_ignition_plugins.msgs
+/// \interface Doppler Velocity Log Tracking message
+/// \brief Velocity tracking message for a Doppler Velocity Log. Doppler velocity
+/// logs are used by the maritime community to track the velocity of a vessel.
+/// It relies on the principle of doppler shift of an echo to measure the velocity
+/// of the craft.
+
+import "ignition/msgs/header.proto";
+import "lrauv_ignition_plugins/msgs/dvl_beam_state.proto";
+import "lrauv_ignition_plugins/msgs/dvl_tracking_target.proto";
+import "lrauv_ignition_plugins/msgs/dvl_kinematic_estimate.proto";
+
+message DVLVelocityTracking
+{
+  /// \brief Message header
+  Header header = 1;
+
+  enum DVLType
+  {
+    PISTON = 0;
+    PHASED_ARRAY = 1;
+  };
+  /// \brief Type of DVL
+  DVLType type = 2;
+
+  /// \brief Locked on target
+  DVLTrackingTarget target = 3;
```

Comment:
Hmm, here you can only specify one tracking target (i.e. tracking mode). IIUC DVLs can do both simultaneously. We could cope with that using different topics, but we can also refactor this to include both (surely not having to synchronize them downstream would be best).

Reply:
Will we be estimating this at all?

Reply:
Well, there will always be a target: either a water mass or the sea floor. The amount of data we'll provide may vary (and thus why several fields are optional).